daemon/cluster/executor/container/container.go

```diff
@@ -374,6 +376,12 @@ func (c *containerConfig) hostConfig() *enginecontainer.HostConfig {
 		}
 	}
 
+	labels := c.task.Networks[0].Network.Spec.Annotations.Labels
+	name := c.task.Networks[0].Network.Spec.Annotations.Name
+
+	if v, ok := labels[naming.PredefinedLabel]; ok && v == "true" && name == "host" {
+		hc.NetworkMode = "host"
```

Comment:
isnt it enough to check for the presence of `PredefinedLabel` and set `hc.NetworkMode=name` ?
That will also satisfy the requirement for `bridge` and `none` networks. Isnt it ?